tagformat/caseconversion/case_conversion.go

```diff
@@ -153,6 +153,50 @@ func DecodeGoCamelCase(s string) (DecodedIdentifier, error) {
 	return words, nil
 }
 
+// DecodeGoTags is similar to DecodeGoCamelCase but slightly more flexible by
+// allowing hyphens (-) in addition to underscores (_) since go struct tags can
+// have these characters. The function will decode CamelCase and kebab-case
+// strings into a slice of lower cased strings.
+func DecodeGoTags(s string) (DecodedIdentifier, error) {
+	words := []string{}
+	lastBoundary := 0
+	for i, char := range s {
+		if firstCharOfInitialism(s, i) || firstCharAfterInitialism(s, i) || char == '_' || char == '-' {
```

Comment:
I'm not sure, but I feel like this is the only line that differs between this and `DecodeGoCamelCase`.
Can we factor out most of the body of these two into a common function that just takes an extra word-boundary callback?
(I think its signature would look like `func(s string, i int, c rune) bool`)

Reply:
Good call. Done